darts/models/forecasting/forecasting_model.py

```diff
@@ -405,6 +419,22 @@ def historical_forecasts(
                 logger,
             )
 
+        if isinstance(retrain, bool) or (isinstance(retrain, int) and retrain >= 0):
+            retrain_func = _retrain_wrapper(
```

Comment:
In this case, could we bypass the call to `retrain_func` altogether when deciding to retrain?
I think it would be slightly better in some cases for performance reasons, because each call to `retrain_func()` requires some (potentially) relatively expensive calls to `drop_after()` on some series, whereas when `retrain` is bool or int we wouldn't need to produce these time series.

Reply:
@FBruzzesi did you get a chance yet to work on these last remaining small changes?

Reply:
I probably didn't mentioned it anywhere. In line 441 I added a variable containing the original retrain function signature. Then in the control flow I am passing the past/future series (hence calling `drop_after` method) only if it is contained in such variable. If it's not, then I just pass `None`